apps/app/pages/[workspaceSlug]/projects/[projectId]/pages/[pageId].tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useState } from "react";
+
+import { useRouter } from "next/router";
+
+import useSWR, { mutate } from "swr";
+
+// lib
+import { requiredAuth } from "lib/auth";
+
+// services
+import projectService from "services/project.service";
+
+// layouts
+import AppLayout from "layouts/app-layout";
+// ui
+import { BreadcrumbItem, Breadcrumbs } from "components/breadcrumbs";
+
+// fetching keys
+import { PAGE_BLOCK_LIST, PROJECT_DETAILS } from "constants/fetch-keys";
+// components
+import { CustomMenu } from "components/ui";
+
+// types
+import { IPageBlock, IView } from "types";
+import type { NextPage, GetServerSidePropsContext } from "next";
+import pagesService from "services/pages.service";
+import useToast from "hooks/use-toast";
+
+const PageBlock: React.FC<any> = ({ pageBlock }: { pageBlock: IPageBlock }) => {
+  const [name, setName] = useState(pageBlock.name);
+  const { setToastAlert } = useToast();
+  const {
+    query: { workspaceSlug, projectId, pageId },
+  } = useRouter();
+
+  const updatePageBlock = async () => {
+    const pageBlockId = pageBlock.id;
+    await pagesService
+      .patchPageBlock(
+        workspaceSlug as string,
+        projectId as string,
+        pageId as string,
+        pageBlockId as string,
+        {
+          name,
+        }
+      )
+      .then(() => {
+        mutate(PAGE_BLOCK_LIST(pageId as string));
+        console.log("Updated block");
+      })
+      .catch(() => {
+        setToastAlert({
+          type: "error",
+          title: "Error!",
+          message: "Page could not be updated. Please try again.",
+        });
+      });
+  };
+
+  const deletePageBlock = async () => {
+    const pageBlockId = pageBlock.id;
+    await pagesService
+      .deletePageBlock(
+        workspaceSlug as string,
+        projectId as string,
+        pageId as string,
+        pageBlockId as string
+      )
+      .then(() => {
+        mutate(PAGE_BLOCK_LIST(pageId as string));
+        console.log("deleted block");
+      })
+      .catch(() => {
+        setToastAlert({
+          type: "error",
+          title: "Error!",
+          message: "Page could not be deleted. Please try again.",
+        });
+      });
+  };
+
+  return (
+    <li className="group relative list-item rounded p-2 pr-6 hover:bg-slate-100">
+      <input
+        type="text"
+        value={name}
+        onKeyDown={(e) => {
+          if (e.key === "Enter") {
+            console.log("Updating...");
+            updatePageBlock();
+          }
+        }}
+        onChange={(e) => {
+          setName(e.target.value);
+        }}
+        className="border-none bg-transparent outline-none"
+      />
+
+      <div className="absolute top-[50%] right-0 z-20 mr-2 hidden translate-y-[-50%] group-hover:block">
```

Comment:
We don't need to use absolute positioning, just flex justify between would be fine

Reply:
yeah I have fixed it